breathecode/payments/views.py

```diff
@@ -996,6 +995,36 @@ def get(self, request, subscription_id=None, academy_id=None):
 
         return handler.response(serializer.data)
 
+    def put(self, request, subscription_id, academy_id=None):
+        lang = get_user_language(request)
+
+        if not (subscription := Subscription.objects.filter(id=subscription_id).first()):
+            raise ValidationException(
+                translation(lang, en="Subscription not found", es="No existe la suscripción", slug="not-found"),
+                code=404,
+            )
+
+        valid_statuses = [choice[0] for choice in Subscription._meta.get_field("status").choices]
+
+        allowed_fields = ["status", "valid_until", "plan"]
+        for field, value in request.data.items():
```

Comment:
The request could be a list or a list of dicts, you could separate the logic in a function within put and check that using `isinstance(request.data, list)`